brightwind/transform/transform.py

```diff
@@ -1400,3 +1401,282 @@ def offset_timestamps(data, offset, date_from=None, date_to=None, overwrite=Fals
+        data_dev_orient_adj = bw.apply_device_orientation_offset(data, fl1, wdir_cols=['Dir_250m', 'Dir_200m'])
+
+    ::
+    Adjust all wind directions by the device orientation, applying inplace:
+        bw.apply_device_orientation_offset(data, fl1, inplace=True)
+        print('Wind direction device orientation offset adjustment is completed.')
+    
+    """
+    
+    measurements = meas_station_data_models.measurements
+    wdirs_properties = _get_consistent_properties_format(measurements, 'wind_direction')
+    # copy the data if needed
+    data = data.copy(deep=True) if inplace is False else data
+    wdir_not_in_dataset = False
+    col_not_in_data = []
+    col_not_in_datamodel = []
+    df = pd.DataFrame(data) if isinstance(data, pd.Series) else data
+
+    if wdir_cols:
+        wdirs_properties_temp = []
+        for col in wdir_cols:
+            if col not in df.columns:
+                wdir_not_in_dataset = True
+                col_not_in_data.append(col)
+            if not any(col == prop['name'] for prop in wdirs_properties):
+                col_not_in_datamodel.append(col)
+            else:
+                # keep wind direction properties only for input wdir_cols
+                wdirs_properties_temp.extend([prop for prop in wdirs_properties if col == prop['name']])
+        wdirs_properties = wdirs_properties_temp
+
+    _check_vertical_profiler_properties_not_overlap(meas_station_data_models, df)
+
+    i = 0
+    # Apply the offset
+    for wdir_prop in wdirs_properties:
+        name = wdir_prop['name']
+        if name in df.columns:
+            date_to = wdir_prop.get('date_to')
+            date_from = wdir_prop.get('date_from')
+            date_from = (df.index[0].strftime('%Y-%m-%dT%H:%M:%S') 
+                         if date_from is None or date_from == DATE_INSTEAD_OF_NONE else date_from)
+            logger_offset = wdir_prop.get('logger_measurement_config.offset')
+            for meas_station_data_model in meas_station_data_models:
+                meas_station_data_model_from = meas_station_data_model.get('date_from')
+                meas_station_data_model_from = (df.index[0].strftime('%Y-%m-%dT%H:%M:%S') if
+                                                meas_station_data_model_from is None or meas_station_data_model_from ==
+                                                DATE_INSTEAD_OF_NONE else meas_station_data_model_from)
+                meas_station_data_model_to = meas_station_data_model.get('date_to')
+                
+                if date_to is None or date_to == DATE_INSTEAD_OF_NONE:
+                    date_to_tmp = meas_station_data_model_to
+                else:
+                    idx_pos = df.index.get_indexer([pd.Timestamp(date_to)], method='nearest')[0]
+                    date_to_tmp = df.index[idx_pos + 1].strftime('%Y-%m-%dT%H:%M:%S') if idx_pos + 1 < len(df.index) else date_to
+
+                if i == 0 or previous_name != name:
```

Comment:
PyCharm is giving this issue
![image](https://github.com/user-attachments/assets/4d77b43b-b9c8-441f-84f0-c272ef7f8400)